### ISSUE TITLE
Question about metrics

### ISSUE THREAD

**Issue by AmirAliSobhGol** [linkedin/kafka-monitor#215]
Hi, i have a question about the following metrics:
kmf_services_consume_service_single_cluster_monitor_records_delay_ms_avg 
kmf_services_produce_service_single_cluster_monitor_produce_delay_ms_99th

my question is how are these metrics are being calculated? 
is it the average record delay of all messages produced by the monitor until it is restarted? or is it the avg record delay of records produced, say every second?

basically, my question is that what is the time quantile of the reported metrics? (avg over 1min? 1second etc.)

also is it possible to expose these metrics in their, kinda raw form? such as a cumulative histogram?

and do u have any suggestion for calculating SLOs over a month? it seems to me given the current metrics the only way is to average, averages and average 99th percentiles which are both wrong.

**Comment by AmirAliSobhGol** [comment]
@andrewchoi5 
thanks, but i think you misunderstood my question, i can understand what the metric is, what i cannot understand is the aggregation part. over what time period are the metrics being aggregated?

**Comment by andrewchoi5** [comment]
For production, the metrics are calculated with the following formula 
```
if (recordsProduced + produceError > 0) {
              availabilitySum += recordsProduced / (recordsProduced + produceError);
```
For consumption, the metrics are calculated with the following formula:
```
       recordsConsumedRate + recordsLostRate > 0
            ? (recordsConsumedRate - recordsDelayedRate) / (recordsConsumedRate + recordsLostRate) : 0;
```
Let's take Produce Service's Scheduled Executor Service for instance. The ProduceExecutor schedules tasks with a FixedDelay on a ProduceRunnable. The initial delay is a configuration value and the running interval is also a configuration value that you set your application to.

`    _produceDelayMs = config.getInt(ProduceServiceConfig.PRODUCE_RECORD_DELAY_MS_CONFIG);`
This config is called `produce.record.delay.ms`.

As for the `records_delay_ms_avg`:

`_recordsDelay` is defined as the mean latency of messages or records from Xinfra Monitor producer to Xinfra Monitor consumer.

sensors records the recordsDelay by calculating the difference between current time in milliseconds (system time) and the previous time in milliseconds, where the prevMs is avroRecord's time stamp. Once your Xinfra Monitor application and the service starts running from the bootup, this will continuously execute and that's how the mean latency of produce to consume occurs.
